lib/active_record_shards/connection_switcher.rb

```diff
@@ -11,6 +11,19 @@ def self.extended(base)
 
       base.singleton_class.send(:alias_method, :table_exists_without_default_shard?, :table_exists?)
       base.singleton_class.send(:alias_method, :table_exists?, :table_exists_with_default_shard?)
+
+      base.singleton_class.send(:alias_method, :connection_without_slave_schema_cache, :connection)
+      base.singleton_class.send(:alias_method, :connection, :connection_with_slave_schema_cache)
+    end
+
+    # we share a single schema-cache across all connections, so columns etc queries get cached
+    # this is safe as long as all connections have the same schema
+    # TODO: make this the slave connections schema_cache for even more savings
+    def connection_with_slave_schema_cache
+      c = connection_without_slave_schema_cache
+      @schema_cache ||= c.schema_cache
+      c.instance_variable_set(:@schema_cache, @schema_cache)
```

Comment:
Isn’t there a `#schema_cache=` method that we can use? https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L153-L156

Reply:
not in 3.2 ... so using the universal way ...